test/Common/TestHelpers.cs

```
@@ -353,9 +353,19 @@ public static async Task WaitUntilTrue(Func<bool> predicate, string conditionDes
 
             do
             {
-                if (predicate())
+                try
                 {
-                    return;
+                    if (predicate())
+                    {
+                        return;
+                    }
+                }
+                catch (Exception ex)
```

With this change we now swallow exceptions and let the test continue running, assuming they are transient. Is this the intended behavior? If so, can you add comments explaining what kind of transient errors this is meant to handle?

---

This change was needed to prevent a newly-exposed race condition in RemovesNewlinesFromExceptions from failing the tests when the LinuxAppServiceLogger was still buffering the file output when the test went to read the file, leading to an equivalent to FileNotFound in our complex read-safe log file reader. 
I can add a comment about this, but I think that tests where the predicate fails due to a determinate exception will continue to fail, so the effect will be
1. These tests will start failing due to timeout errors, with the "real" exception details in the logs
2. Non-determinate/race-condition failures will be swallowed. 

I'll add a comment explaining the same.